### PR TITLE
Add Skip Import Without Name feature

### PR DESCRIPTION
**Pull request by Dhairya-Rupala**
https://user-images.githubusercontent.com/77922322/171860116-bb13b6e1-d0bd-42d7-97e3-535f1501d47a.mp4

 **Details** 

- **By default the Skip Unnamed Chunks checkbox is unselected** 

>1. So it will work as it was working before.it will list all the chunks including chunks with name and without name.
>2. User will have two options, copy chunks with files or copy chunks without files.
>3. When user saves the chunks all the chunks will be saved.

- **If the checkbox is selected then**

>1. All the imports with no chunk name specified will be skipped.
>2. In this case the default copy button will copy only the filtered chunks.
>3. When user saves the chunks then only the chunks with names will be saved.

- **Side navigation bar issue resolved**
> Issue : When we open the site the side navigation bar icon tabs are being rendered but they were not displayed on the browser, as they were being covered by the other elements, this was because material UI CSS was overlapping the predefined CSS. Now it has been resolved.

https://user-images.githubusercontent.com/77922322/172781365-7ee2d815-919d-470e-942f-deac3aa6d637.mov

https://user-images.githubusercontent.com/77922322/172781548-d0016885-ce0e-4a8c-bfb9-9c182b50d36f.mov

- **Added Tool-tips for Icons**

Screenshot 2022-06-29 at 6 33 45 PM

### CODE REVIEW

File: components/ChunksPicker.js [hunk 1]
```diff
@@ -28,6 +30,7 @@ import SaveIcon from '@material-ui/icons/Save';
 import { HomeOutlined } from '@material-ui/icons';
 import { useQuery } from 'react-query';
 import SaveCollectionForm from '../components/SaveCollectionForm';
+import { getLocationOrigin } from 'next/dist/shared/lib/utils';
```
Comment (explorer02) [inline components/ChunksPicker.js]: remove

File: components/ChunksPicker.js [hunk 2]
```diff
@@ -62,13 +65,22 @@ const createPostReqOptions = (obj) => {
     body: JSON.stringify(obj),
   };
 };
+
+
+// Chunks picker
```
Comment (explorer02) [inline components/ChunksPicker.js]: remove

File: components/ChunksPicker.js [hunk 3]
```diff
@@ -62,13 +65,22 @@ const createPostReqOptions = (obj) => {
...
 const ChunksPicker = ({ entryFile, className }) => {
   const classes = useStyles();
+
+  // checkBox for skipping the import without the name
```
Comment (explorer02) [inline components/ChunksPicker.js]: remove

File: components/ChunksPicker.js [hunk 4]
```diff
@@ -62,13 +65,22 @@ const createPostReqOptions = (obj) => {
...
+  const [skipBox, setSkipBox] = useState(false);
```
Comment (explorer02) [inline components/ChunksPicker.js]: [skipImportWithoutChunkName, setSkipImportWithoutChunkName]

File: components/ChunksPicker.js [hunk 5]
```diff
@@ -62,13 +65,22 @@ const createPostReqOptions = (obj) => {
...
+
+
+
   // loads all descendent chunks
   const loadAllDescendantChunks = useCallback(
     (filepath) =>
       new Promise((resolve, reject) => {
-        fetch('api/chunks', createPostReqOptions({ path: filepath, getDescendant: true }))
+        fetch('api/chunks', createPostReqOptions({ path: filepath, getDescendant: true,skipImportWithoutChunkName:skipBox })) // request
```
Comment (explorer02) [inline components/ChunksPicker.js]: remove comment

File: components/ChunksPicker.js [hunk 6]
```diff
@@ -100,14 +112,14 @@ const ChunksPicker = ({ entryFile, className }) => {
     if (!path) {
       return [];
     }
-    const response = await fetch('api/chunks', createPostReqOptions({ path: path, getDescendant: false }));
+    const response = await fetch('api/chunks', createPostReqOptions({ path: path, getDescendant: false,skipImportWithoutChunkName:skipBox }));
```
Comment (explorer02) [inline components/ChunksPicker.js]: space in between

File: components/ChunksPicker.js [hunk 7]
```diff
@@ -100,14 +112,14 @@ const ChunksPicker = ({ entryFile, className }) => {
...
     const data = await response.json();
     const chunkWithName = data.chunks;
     return chunkWithName;
   };
 
   const { isLoading: isLoadingChunks, data: childrenChunks } = useQuery(
-    'getChunks' + crumbs[crumbs.length - 1].filepath,
+    'getChunks' + crumbs[crumbs.length - 1].filepath + skipBox,
```
Comment (explorer02) [inline components/ChunksPicker.js]: 'getChunks' + crumbs[crumbs.length - 1].filepath + (skip?'-skip':''),

File: components/ChunksPicker.js [hunk 8]
```diff
@@ -137,6 +149,13 @@ const ChunksPicker = ({ entryFile, className }) => {
   selectedChunksRef.current = selectedChunks;
   processingRef.current = processing;
 
+
+  //hnadling skip button 
+  const handleSkipBox = (e) => {
```
Comment (explorer02) [inline components/ChunksPicker.js]: toggleSkipImportButton,
useCallback

File: components/ChunksPicker.js [hunk 9]
```diff
@@ -243,6 +262,29 @@ const ChunksPicker = ({ entryFile, className }) => {
   const [shouldShowSnackbar, setSnackbarVisibility] = useState(false);
   const snackBarMessage = useRef('');
   const hideSnackbar = useCallback(() => setSnackbarVisibility(false), []);
+
+
+
+  // handler for filtered copy, filters the chunks without name
+  const handleFilteredCopy = () => {
```
Comment (explorer02) [inline components/ChunksPicker.js]: useCallback

File: components/ChunksPicker.js [hunk 10]
```diff
@@ -243,6 +262,29 @@ const ChunksPicker = ({ entryFile, className }) => {
...
+    const extensions = ['.js', '.ts', '.tsx'];
```
Comment (explorer02) [inline components/ChunksPicker.js]: export from getAllChunks

File: components/ChunksPicker.js [hunk 11]
```diff
@@ -243,6 +262,29 @@ const ChunksPicker = ({ entryFile, className }) => {
...
+    const displayedChunks = Array.from(selectedChunks);
+    const filteredChunks = displayedChunks.filter((chunk) => {
+      for (let ext of extensions) {
```
Comment (explorer02) [inline components/ChunksPicker.js]: Array.some()

File: components/ChunksPicker.js [hunk 12]
```diff
@@ -310,7 +352,17 @@ const ChunksPicker = ({ entryFile, className }) => {
     setIsChildrenChunks(childrenChunks != null && childrenChunks.length > 0);
   }, [childrenChunks]);
 
+
+  
+
   return !!crumbs[crumbs.length - 1]?.filepath || !!selectedChunks.size ? (
+    <>
+      {/* Skip button for without name imports */}
+      <FormControlLabel
```
Comment (explorer02) [inline components/ChunksPicker.js]: check if can be rendered in box below

File: components/ChunksPicker.js [hunk 13]
```diff
@@ -369,7 +421,16 @@ const ChunksPicker = ({ entryFile, className }) => {
             />
 
             {/*Buttons to perform delete or copy on above TextField, save collection for selected chunks */}
-            <Box ml="auto">
+              <Box ml="auto">
+                
+                {
+                  skipBox?<></>:<>
```
Comment (explorer02) [inline components/ChunksPicker.js]: skipBox?null: Component

File: components/ChunksPicker.js [hunk 14]
```diff
@@ -369,7 +421,16 @@ const ChunksPicker = ({ entryFile, className }) => {
...
+                  <IconButton disabled={!selectedChunks.size} onClick={handleFilteredCopy} aria-label="Filtered copy">
```
Comment (explorer02) [inline components/ChunksPicker.js]: copy without files

File: components/ChunksPicker.js [hunk 15]
```diff
@@ -385,6 +446,7 @@ const ChunksPicker = ({ entryFile, className }) => {
               >
                 <SaveIcon />
               </IconButton>
+
```
Comment (explorer02) [inline components/ChunksPicker.js]: extra line?

File: package.json [hunk 16]
```diff
@@ -31,7 +31,7 @@
     "fuzzy-search": "^3.2.1",
     "glob": "^7.1.7",
     "glob-promise": "^4.2.0",
-    "next": "12.1.6",
+    "next": "^12.1.6",
```
Comment (explorer02) [inline package.json]: revert this file changes

File: pages/api/chunks.ts [hunk 17]
```diff
@@ -7,16 +7,21 @@ type Data = {
   pages: any;
 };
 
+
+
 export default async function handler(req: any, res: NextApiResponse<Data>) {
```
Comment (explorer02) [inline pages/api/chunks.ts:10]: extra lines

File: utils/getAllChunks.ts [hunk 18]
```diff
@@ -4,25 +4,48 @@ import { File } from '@babel/types';
...
+
+
+export const getAllChunks = (path: string, root: string, getDescendant: boolean = false,skipImportWithoutChunkName:boolean): Record<string, any> => {
+
+  //resolving path to absolute path 
   path = resolve(root, path);
+
+
   const descendant = getDescendant ? 'deep' : 'shallow';
+  const skipLabel = skipImportWithoutChunkName ? "skip" : "noskip";
```
Comment (explorer02) [inline utils/getAllChunks.ts]: better name, store in variable

File: utils/getAllChunks.ts [hunk 19]
```diff
@@ -31,21 +54,25 @@ export const getAllChunks = (path: string, root: string, getDescendant: boolean
     });
   }
 
-  if (store[descendant][path] !== undefined) {
-    return Promise.resolve(store[descendant][path]);
+  //if already calculated then return it
+  if (store[descendant][skipLabel][path] !== undefined) {
```
Comment (explorer02) [inline utils/getAllChunks.ts]: explain

File: utils/getAllChunks.ts [hunk 20]
```diff
@@ -162,8 +194,11 @@ const getImports = (ast: File) => {
           }
         }
 
-        chunkPathToName[filePath] = undefined;
-        dynamicImports.add(filePath);
+        if (skipImportWithoutChunkName === false)
```
Comment (explorer02) [inline utils/getAllChunks.ts]: if(!skip...)

File: components/ChunksPicker.js [hunk 21]
```diff
@@ -28,6 +26,14 @@ import SaveIcon from '@material-ui/icons/Save';
 import { HomeOutlined } from '@material-ui/icons';
 import { useQuery } from 'react-query';
 import SaveCollectionForm from '../components/SaveCollectionForm';
+import { extensions } from "../pages/extensions.tsx";
```
Comment (explorer02) [inline components/ChunksPicker.js]: remove extension

File: components/ChunksPicker.js [hunk 22]
```diff
@@ -62,13 +68,16 @@ const createPostReqOptions = (obj) => {
...
   const classes = useStyles();
-  // loads all descendent chunks
+
+  const [skipImportWithoutChunkName, setSkipImportWithoutChunkName] = useState(false);
+
   const loadAllDescendantChunks = useCallback(
     (filepath) =>
       new Promise((resolve, reject) => {
-        fetch('api/chunks', createPostReqOptions({ path: filepath, getDescendant: true }))
+        fetch('api/chunks', createPostReqOptions({ path: filepath, getDescendant: true, skipImportWithoutChunkName:skipImportWithoutChunkName }))
```
Comment (explorer02) [inline components/ChunksPicker.js]: no need for alias 
createPostReqOptions({ path: filepath, getDescendant: true, skipImportWithoutChunkName }))

File: components/ChunksPicker.js [hunk 23]
```diff
@@ -100,16 +109,17 @@ const ChunksPicker = ({ entryFile, className }) => {
     if (!path) {
       return [];
     }
-    const response = await fetch('api/chunks', createPostReqOptions({ path: path, getDescendant: false }));
+    const response = await fetch('api/chunks', createPostReqOptions({ path: path, getDescendant: false, skipImportWithoutChunkName:skipImportWithoutChunkName }));
```
Comment (explorer02) [inline components/ChunksPicker.js]: remove alias

File: components/ChunksPicker.js [hunk 24]
```diff
@@ -137,6 +147,13 @@ const ChunksPicker = ({ entryFile, className }) => {
   selectedChunksRef.current = selectedChunks;
   processingRef.current = processing;
 
+
+  //hnadling skip button
```
Comment (explorer02) [inline components/ChunksPicker.js]: typo

File: components/ChunksPicker.js [hunk 25]
```diff
@@ -243,11 +260,23 @@ const ChunksPicker = ({ entryFile, className }) => {
   const [shouldShowSnackbar, setSnackbarVisibility] = useState(false);
   const snackBarMessage = useRef('');
   const hideSnackbar = useCallback(() => setSnackbarVisibility(false), []);
+
+
+
+
```
Comment (explorer02) [inline components/ChunksPicker.js]: too many extra lines, keep 1-2 at max

File: components/ChunksPicker.js [hunk 26]
```diff
@@ -243,11 +260,23 @@ const ChunksPicker = ({ entryFile, className }) => {
   const [shouldShowSnackbar, setSnackbarVisibility] = useState(false);
   const snackBarMessage = useRef('');
   const hideSnackbar = useCallback(() => setSnackbarVisibility(false), []);
+
+
+
+  
+  // handler for filtered copy, filters the chunks without name
+  const handleFilteredCopy = useCallback(() => {
```
Comment (explorer02) [inline components/ChunksPicker.js]: handleCopyImportWithoutChunk

File: components/ChunksPicker.js [hunk 27]
```diff
@@ -243,11 +260,23 @@ const ChunksPicker = ({ entryFile, className }) => {
...
+    const filteredChunks = removeUnnamedChunks([...selectedChunks]);
+    snackBarMessage.current = `${filteredChunks.length} chunks copied`;
+    navigator.clipboard.writeText(filteredChunks.join()).then(() => setSnackbarVisibility(true));
+  },[selectedChunks]);
+
+  const adaptedChunks = skipImportWithoutChunkName ? (removeUnnamedChunks([...selectedChunks])) : ([...selectedChunks]);
```
Comment (explorer02) [inline components/ChunksPicker.js]: useMemo

File: components/ChunksPicker.js [hunk 28]
```diff
@@ -311,7 +340,13 @@ const ChunksPicker = ({ entryFile, className }) => {
   }, [childrenChunks]);
 
   return !!crumbs[crumbs.length - 1]?.filepath || !!selectedChunks.size ? (
-    <Box mt={2} className={className} display="flex" flexDirection="column">
+      <Box mt={2} className={className} display="flex" flexDirection="column">
+      {/* Skip button for without name imports */}
+        <FormControlLabel style={{ marginBottom: '11px', width:"fit-content"}}
```
Comment (explorer02) [inline components/ChunksPicker.js]: try ml={3} for margin

File: components/ChunksPicker.js [hunk 29]
```diff
@@ -369,15 +404,23 @@ const ChunksPicker = ({ entryFile, className }) => {
             />
 
             {/*Buttons to perform delete or copy on above TextField, save collection for selected chunks */}
-            <Box ml="auto">
-              <IconButton disabled={!selectedChunks.size} onClick={handleCopy} aria-label="copy">
+              <Box ml="auto">
+                {/* Filter copy button, only copies the chunks with chunk name */}
+                {
+                  skipImportWithoutChunkName?null:<>
```
Comment (explorer02) [inline components/ChunksPicker.js]: add space between ?, :

File: components/ChunksPicker.js [hunk 30]
```diff
@@ -458,7 +501,7 @@ const ChunksPicker = ({ entryFile, className }) => {
                 top="0"
                 overflow="auto"
               >
-                {[...selectedChunks].map((chunk) => (
+                {[...adaptedChunks].map((chunk) => (
```
Comment (explorer02) [inline components/ChunksPicker.js]: might not need to spread

File: components/Layout.js [hunk 31]
```diff
@@ -51,7 +51,7 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 const routesToName = {
-  '/': 'Selecter',
+  '/': 'Selector',
```
Comment (explorer02) [inline components/Layout.js:54]: why this change?

File: pages/extensions.tsx [hunk 32]
```diff
@@ -0,0 +1 @@
+export const extensions = ['.js', '.ts', '.tsx', '/index.js', '/index.ts', '/index.tsx'];
```
Comment (explorer02) [inline pages/extensions.tsx]: move this file to constants/extensions.ts

File: components/ChunksPicker.js [hunk 33]
```diff
@@ -25,9 +23,18 @@ import FileCopyOutlinedIcon from '@material-ui/icons/FileCopyOutlined';
 import DeleteOutlineIcon from '@material-ui/icons/DeleteOutline';
 import Chip from '@material-ui/core/Chip';
 import SaveIcon from '@material-ui/icons/Save';
+import { Tooltip } from '@material-ui/core';
 import { HomeOutlined } from '@material-ui/icons';
 import { useQuery } from 'react-query';
 import SaveCollectionForm from '../components/SaveCollectionForm';
+import { extensions } from "../constants/extensions";
+
+const removeUnnamedChunks = function (chunks) {
```
Comment (explorer02) [inline components/ChunksPicker.js]: arrow function

File: components/ChunksPicker.js [hunk 34]
```diff
@@ -243,11 +260,23 @@ const ChunksPicker = ({ entryFile, className }) => {
   const [shouldShowSnackbar, setSnackbarVisibility] = useState(false);
   const snackBarMessage = useRef('');
   const hideSnackbar = useCallback(() => setSnackbarVisibility(false), []);
+
+  // handler for filtered copy, filters the chunks without name
+  const handleCopyImportWithChunk = useCallback(() => {
+    const filteredChunks = removeUnnamedChunks([...selectedChunks]);
```
Comment (explorer02) [inline components/ChunksPicker.js:265]: removeUnnamedChunks(selectedChunks)

File: components/ChunksPicker.js [hunk 35]
```diff
@@ -243,11 +260,23 @@ const ChunksPicker = ({ entryFile, className }) => {
...
+    snackBarMessage.current = `${filteredChunks.length} chunks copied`;
+    navigator.clipboard.writeText(filteredChunks.join()).then(() => setSnackbarVisibility(true));
+  },[selectedChunks]);
+
+  const adaptedChunks = useMemo(() => {
+    return skipImportWithoutChunkName ? (removeUnnamedChunks([...selectedChunks])) : ([...selectedChunks]);
```
Comment (explorer02) [inline components/ChunksPicker.js:271]: removeUnnamedChunks(selectedChunks)

File: yarn.lock [hunk 36]
```diff
@@ -2,120 +2,120 @@
 # yarn lockfile v1
```
Comment (explorer02) [inline yarn.lock]: remove this file

**Review by chinmaychaudhary — APPROVED** [review]
